src/tc-yaml.js

```diff
@@ -252,18 +252,24 @@ class VersionOne extends TcYaml {
       }
 
       config.tasks = config.tasks.map(task => {
-        task = _.defaults(task, {
+        task.routes = task.routes || [];
+        config.reporting && task.routes.push(cfg.app.checkTaskRoute);
+
+        task = Object.assign({
           taskId: defaultTaskId,
           taskGroupId: defaultTaskGroupId,
           created: now,
-        });
+        }, task);
         defaultTaskId = slugid.nice(); // invent a new taskId for the next task
 
+        const {taskId, ...taskWithoutTaskId} = task;
+        console.log(task.routes);
         return {
-          taskId: task.taskId,
-          task: _.omit(
-            _.extend(task, {schedulerId: cfg.taskcluster.schedulerId}),
-            'taskId'),
+          taskId,
+          task: {
+            ...taskWithoutTaskId,
+            schedulerId: cfg.taskcluster.schedulerId,
```

Comment:
I don't think they _must_. This code doesn't care about scheduler id because we have  more precise filtering. However, I think having it in the task definition won't hurt, and might be useful. You think it's better not to have it anymore?

Reply:
Yeah, I think we used to override it because we used it for filtering before. I decided not to change this at this point